icicle/tests/polynomial_test.cu

```diff
@@ -342,18 +318,18 @@ TEST_F(PolynomialTest, ReadCoeffsToHost)
   auto g = Polynomial_t::from_coefficients(coeffs_g, 3);
 
   auto h = f + g; // 1+2x+3x^3
-  const auto h0 = h.copy_coefficient_to_host(0);
-  const auto h1 = h.copy_coefficient_to_host(1);
-  const auto h2 = h.copy_coefficient_to_host(2);
+  const auto h0 = h.get_coeff(0);
+  const auto h1 = h.get_coeff(1);
+  const auto h2 = h.get_coeff(2);
   EXPECT_EQ(h0, one);
   EXPECT_EQ(h1, two);
   EXPECT_EQ(h2, three);
 
-  int64_t nof_coeffs = h.copy_coefficients_to_host(nullptr); // query #coeffs
-  EXPECT_GE(nof_coeffs, 3);                                  // can be larger due to padding to powers of two
+  int64_t nof_coeffs = h.copy_coeffs(nullptr); // query #coeffs
```

Comment:
it's returning nof_coeffs without copying when seeing nullptr. It's so that I can allocate the right amount of memory and call again if I want.

Reply:
looks quite cryptic - maybe some function if called elsewhere like that or at least comment?

Reply:
I agree it's not super elegant but: the intention was that a user calls with null, then based on the return value allocates memory and calls again.

- there is a degree() function that can be called to check how many coeffs there really are. 
- I could add another function to get the memory size (which can be larger than degree because I need powers of two memory size for NTTs) but I would rather not do that.
- note that copy_coeffs() is returning the #elements that were copied anyway and that if the input pointer is nullptr than I cannot copy.
- This is similar to CUB library that can be called with nullptrs and returns the size of memory that should be alllocated

Now that I think about it, if a user calls degree(), he may allocate a smaller memory than I internally have (for example for degree 10 I may have 16 elements internally) and this would cause a memcpy issue if he tells me start=0, end=-1. So maybe I should remove the default values for start/end_idx. A user must know the size anyway because he allocates the memory. If he doesn't, he can call degree().

I will do that, but still what should I do with nullptr? throw error? what do you think?

@vhnatyk